src/libraries/Microsoft.PowerFx.Core/Public/Marshal/PrimitiveMarshallerProvider.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx.Core
+{
+    public class PrimitiveMarshallerProvider : ITypeMashallerProvider
+    {
+        // Map from .net types to formulaTypes
+        private static readonly Dictionary<Type, FormulaType> _map = new Dictionary<Type, FormulaType>()
+        {
+            // Fx needs more number typeS:
+            { typeof(double), FormulaType.Number },
```

Comment:
If we do go add more number types, changing this could be a breaking change in how existing uses marshal types. Just a note, not actionable.

Reply:
Agreed. Although this is just capturing the existing semantics we had in FormulaValue.New.